ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift

```diff
@@ -81,21 +128,44 @@ protocol AnyDownloadBatch {
  * Represents one overall resource-related command for requests against the Keyman Cloud API.
  */
 class DownloadBatch<FullID: LanguageResourceFullID>: AnyDownloadBatch where FullID.Resource.Package: TypedKeymanPackage<FullID.Resource> {
+  typealias CompletionHandler = ResourceDownloadManager.CompletionHandler
+
   public final var downloadTasks: [DownloadTask<FullID>]
   var errors: [Error?] // Only used by the ResourceDownloadQueue.
   public final var startBlock: (() -> Void)? = nil
-  public final var completionBlock: ResourceDownloadManager.CompletionHandler<FullID.Resource>? = nil
+  public final var completionBlock: CompletionHandler<FullID.Resource>? = nil
   
   public init?(do tasks: [DownloadTask<FullID>],
                startBlock: (() -> Void)? = nil,
-               completionBlock: ResourceDownloadManager.CompletionHandler<FullID.Resource>? = nil) {
+               completionBlock: CompletionHandler<FullID.Resource>? = nil) {
     self.downloadTasks = tasks
 
     self.errors = Array(repeating: nil, count: tasks.count)
     self.startBlock = startBlock
     self.completionBlock = completionBlock
   }
 
+  public init(forPackageWithID fullID: FullID,
+              from url: URL,
+              startBlock: (() -> Void)?,
+              completionBlock: CompletionHandler<FullID.Resource>?) {
+    // If we can't build a proper DownloadTask, we can't build the batch.
+    let tempArtifact = ResourceFileManager.shared.packageDownloadTempPath(forID: fullID)
+    let finalFile = ResourceFileManager.shared.cachedPackagePath(forID: fullID)
+
+    let task = DownloadTask(forPackageWithID: fullID, from: url, as: finalFile, tempURL: tempArtifact)
+
+
+    self.downloadTasks = [task]
+    self.errors = Array(repeating: nil, count: 1) // We only build the one task.
+
+    self.startBlock = startBlock
+    self.completionBlock = completionBlock
+
+    task.request.userInfo[Key.downloadTask] = task
+    task.request.userInfo[Key.downloadBatch] = self
+  }
+
```

Comment:
This block is designed mostly as a refactor of: https://github.com/keymanapp/keyman/blob/9761d944478f7ebf2791d7a6df48e11be4fa5ac1/ios/engine/KMEI/KeymanEngine/Classes/Resource%20Management/ResourceDownloadManager.swift#L275-L303

Note that the "source" is not being removed at this time, as the PR already has numerous-enough changes as things already are.  A followup PR will perform the cleanup necessary to remove the original version.